students/tim_lurvey/lesson05/mailroom_pt3.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env python
+__author__ = 'Tim Lurvey, ig408c'
+
+import sys
+# from lurvey_classes import bunch
+class bunch(object):
+    def __init__(self,**kwargs):
+        setattr(self, '__dict__', kwargs)
+
+my_data = {}
+my_data.update({'Tom Hanks': bunch(total=24536.20, number=3)})
+my_data.update({'Barry Larkin': bunch(total=4521., number=3)})
+my_data.update({'Mo Sizlack': bunch(total=88.88, number=2)})
+my_data.update({'Anonymous': bunch(total=100., number=1)})
+my_data.update({'Donnald Trump': bunch(total=1., number=3)})
+
+
+def add_new_name(name: str = ""):
+    my_data.update({name: bunch(total=0., number=0)})
+    return True
+
+
+def compose_email(name: str = "", donation: float = 0., data: bunch = None):
+    """return the string of the formatted email"""
+    if data.number < 1:
+        s = ""
+        is_are = "is"
+    else:
+        s = "s"
+        is_are = "are"
+    #
+    email_str = "\nHello {name},\n\n" \
+                "Thank you for your generous donation of $ {donation:.2f}.\n" \
+                "Your {count} donation{s}, totaling $ {total:.2f}, {is_are} greatly appreciated.\n\n" \
+                "Thank you\n\n".format(name=name,
+                                       donation=donation,
+                                       count=data.number,
+                                       s=s,
+                                       total=data.total,
+                                       is_are=is_are)
+    return email_str
+
+
+def add_donation(name: str = "", ammount: float = 0.):
+    data = my_data.get(name)
+    data.total += float(ammount)
+    data.number += 1
+    my_data.update({name: data})
+    return True
+
+
+def print_name_list():
+    # I am familiar with one-line list comprehension but I know I use it too much
+    # making my code difficult to read, with including lambda functions.  The more
+    # I read and write code, the more I find the need to be verbose.  If speed is
+    # important I will bring the list comprehension into the C level with
+    # one-liners.
+    # "\n".join(["The following names are available:",] + [name for name in my_data.keys()])
```

Comment:
often times one-liners (comprehensions) actually make code more readable because that's what most python devs are used to.

that line can actually be shorter -> 
`"\n".join(["The following names are available:",] + list(my_data))`

I find ^^ version a lot more readable than the 4 lines of code below - (if you really think about it.. reading 4 lines is harder than reading one concise line), but I do understand what you mean.. some people to tend to get really carried away with "one-liners".